src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/config/WebClientConfiguration.kt

```diff
@@ -0,0 +1,86 @@
+package uk.gov.justice.digital.hmpps.prisonersearchindexer.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.scheduling.annotation.EnableAsync
+import org.springframework.security.oauth2.client.AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.ReactiveOAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.ReactiveOAuth2AuthorizedClientProviderBuilder
+import org.springframework.security.oauth2.client.ReactiveOAuth2AuthorizedClientService
+import org.springframework.security.oauth2.client.registration.ReactiveClientRegistrationRepository
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServerOAuth2AuthorizedClientExchangeFilterFunction
+import org.springframework.web.reactive.function.client.WebClient
+
+@Configuration
+@EnableAsync
+class WebClientConfiguration(
+  @Value("\${api.base.url.prison-api}") val prisonApiBaseUri: String,
+  @Value("\${api.base.url.incentives}") val incentivesBaseUri: String,
+  @Value("\${api.base.url.restricted-patients}") val restrictedPatientBaseUrl: String,
+  @Value("\${api.base.url.oauth}") val hmppsAuthBaseUri: String,
+) {
+
+  @Bean
+  fun hmppsAuthHealthWebClient(): WebClient = WebClient.builder().baseUrl(hmppsAuthBaseUri).build()
```

Comment:
If you inject the WebClient.Builder into this method then it comes wired up with the pre-configured Jackson object mapper.

Reply:
(although that is more pertinent to the non-health web clients which don't really need it!)

Reply:
agreed, which is the approach I've taken for the non-health web clients